### PR TITLE
Adds a random wig crate

### PR DESCRIPTION
**Pull request by unfunnyperson**
[CLOTHING] [FEATURE]

## About the PR 
Adds a wig crate for 2000 credits that contains 7 randomized wigs. 

## Why's this needed? 
More things for cargo to buy are always cool. Also allows for bald people, borgs, and mutantraces to get hair more easily.

## Changelog

```changelog
(u)UnfunnyPerson
(+)Added a randomized wig crate. Cargo can order this crate for 2000 credits.
```

### CODE REVIEW

**Comment by TealSeer** [comment]
I love the idea, but the way you have wigs being generated currently leads to some really bizarre ones that I don't know if anybody would want to wear (multiple base hairstyles overlapping, ones that are just the detail modifier of a hairstyle and nothing else, etc.) It might be worth checking out the code used to generate a random character because it has some checks to make sure you only get base hair styles first, then roll a chance to add the details for that hair style. Relevant stuff starts here:

https://github.com/goonstation/goonstation/blob/e39db65de4806ecde98c9ed0ae75ac74344abbba/code/datums/preferences.dm#L2004-L2059

File: code/modules/barber/barber_shop.dm [hunk 1]
```diff
@@ -54,6 +54,25 @@
 		hair_list[third_id] = third_color
 		src.setup_wig(hair_list)
 
+///Randomized wig for the cargo crate
+/obj/item/clothing/head/wig/spawnable/random
+
+	New()
+		var/list/possible_hairstyles = null
```
Comment (TobleroneSwordfish) [inline code/modules/barber/barber_shop.dm:61]:
```suggestion
		src.name = "wig"
		var/list/possible_hairstyles = null
```
they're called toupées by default for some reason

File: code/modules/economy/supply_packs.dm [hunk 2]
```diff
@@ -1187,6 +1187,15 @@ ABSTRACT_TYPE(/datum/supply_packs)
 	containertype = /obj/storage/crate/packing
 	containername = "Haberdasher's Crate"
 
+/datum/supply_packs/random_wigs
+	name = "Wigs"
```
Comment (TobleroneSwordfish) [inline code/modules/economy/supply_packs.dm:1191]:
```suggestion
	name = "Spare wigs crate"
```
To match up with how other crates are named.

**Review by TobleroneSwordfish — CHANGES_REQUESTED** [review]

File: code/modules/barber/barber_shop.dm [hunk 3]
```diff
@@ -54,6 +54,25 @@
...
+		else
+			possible_hairstyles = filtered_concrete_typesof(/datum/customization_style, /proc/isfem)
+		var/datum/customization_style/hair_type = null
+		var/picked_color = rgb(rand(0,255),rand(0,255),rand(0,255))
+		hair_type = pick(possible_hairstyles)
+		first_id = initial(hair_type.id)
+		first_color = picked_color
+		if (prob(33))
+			second_id = initial(hair_type.id)
+			second_color = picked_color
```
Comment (Tarmunora) [inline code/modules/barber/barber_shop.dm:73]: wouldn't you want a *different* second style here

Comment (unfunnyperson) [reply code/modules/barber/barber_shop.dm]: I... how did I miss that :sheltersweat:

**Review by Tarmunora — APPROVED** [review]

**Pull request merged**